packages/core/src/components/Box/react/utils.ts

```diff
@@ -41,23 +42,9 @@ const styleCache: {
   [key: string]: { className: string; css: string } | undefined;
 } = {};
 
-/**
- * Generates a unique key for props, aiding cache efficiency.
- * Incurs O(n log n) cost due to sorting, but it's acceptable given the
- * expensive nature of StyleGenerator's internals.
- */
-// eslint-disable-next-line @typescript-eslint/no-explicit-any -- FIXME
-function generateKey(props: Record<string, any>) {
-  return Object.entries(props)
-    .filter(([, value]) => value !== undefined)
-    .sort((a, b) => a[0].localeCompare(b[0]))
-    .map(([key, value]) => `${key}:${value}`)
-    .join("|");
-}
-
 // eslint-disable-next-line @typescript-eslint/no-explicit-any -- FIXME
 export function getCachedStyle(dynamicProps: Record<string, any>) {
-  const key = generateKey(dynamicProps);
+  const key = stableStringify(dynamicProps);
```

Comment:
Note: I am not sure whether it is intentional to have style cache logic in this file as well